### PR TITLE
Migrate to Jetpack Media3

### PR DESCRIPTION
**Pull request by marcbaechinger**

### CODE REVIEW

**Comment by marcbaechinger** [comment]
@pflammertsma or to whom it may concern :) 

I updated this in the fork to Media3 1.0.0-beta01, so it is included in this PR as well. Looks like a good time now to review/merge this PR into the `media3` branch. IMO we can also merge into the `main` branch as the `MediaSession` API pieces of this app are not too complicated. Up to you regarding the timing of merging into `main` of course.

**Comment by pflammertsma** [comment]
I'm still going through the review, but one curiosity: why have some of the tests in CastHelperTest been removed?

**Comment by marcbaechinger** [comment]
Because the method that is under test there have been removed. The underlying API that we called there is not available for Kotlin anymore.

Please see internal bug ref: b/237992143

**Comment by marcbaechinger** [comment]
I added a commit that changes the life-cycle behaviour as discussed. I made an additional commit on purpose to call this change out so developers can see this specific change to do the right thing.

File: ReferenceAppKotlin/app/src/main/java/com/android/tv/reference/playback/PlaybackFragment.kt [hunk 1]
```diff
@@ -87,77 +86,96 @@ class PlaybackFragment : VideoSupportFragment() {
...
-            it.pause()
-            it.release()
-            exoplayer = null
+    private fun releaseMediaSession() {
+        mediaSession?.run {
+            player.release()
+            release()
+            mediaSession = null
+            // TODO Remove comment after b/218604167 has been release in the Cast TV library.
+            // CastReceiverContext.getInstance().mediaManager.setSessionCompatToken(null)
```
Comment (marcbaechinger) [inline ReferenceAppKotlin/app/src/main/java/com/android/tv/reference/playback/PlaybackFragment.kt:168]: This call replaces the method I have removed from `CastHelper`/`CastHelperTest`. We just call it directly here without the indirection through the `CastHelper`. So we don't need to deal with the bug in the test case. The indirection in `CastHelper` was super trivial and a bit meaningless as it just calls through the cast method. We can put that back when the bug is fixed and we can uncomment the call here.

**Pull request merged**